SEIR/seir.py

```diff
@@ -96,55 +96,48 @@ def steps_SEIR_nb(p_vec, seeding, uid, dt, t_inter, nnodes, popnodes,
     y[S, :] = popnodes
     states = np.zeros((ncomp, nnodes, len(t_inter)))
 
-    mv_one = np.empty(ncomp - 1)
-    mv = np.zeros((ncomp - 1, nnodes))
-    exposeCases = np.empty(nnodes)
-    incidentCases = np.empty(nnodes)
-    incident2Cases = np.empty(nnodes)
-    incident3Cases = np.empty(nnodes)
-    recoveredCases = np.empty(nnodes)
-
     p_infect = 1 - np.exp(-dt * p_vec[1][0][0])
     p_recover = 1 - np.exp(-dt * p_vec[2][0][0])
 
     for it, t in enumerate(t_inter):
-        mv[:] = 0
-        begin = y[:-1]
         if (it % int(1 / dt) == 0):
-            y[I1] = y[I1] + seeding[int(t)]
-            y[cumI] = y[cumI] + seeding[int(t)]
+            y[I1] += seeding[int(t)]
+            y[cumI] += seeding[int(t)]
 
         # calculate matrix of mv's
         # TODO: add all probabilities and do a binomial of that
         for i in range(mobility_len):
-            ori = mobility_ori[i]
-            dest = mobility_dest[i]
-            prob = mobility_prob[i]
             for c in range(ncomp - 1):
-                mv_one[c] = np.random.binomial(y[c, ori], prob)
-            mv[:, ori] -= mv_one
-            mv[:, dest] += mv_one
-
-        y[:-1] += mv
-
-        p_expose = 1 - np.exp(-dt * p_vec[0][it] *
-                              (y[I1] + y[I2] + y[I3]) / popnodes)  # vector
+                delta = np.random.binomial(y[c, mobility_ori[i]], mobility_prob[i])
```

Comment:
The re-write here is consistent w/the code on `master` (which updates the `y` matrix as the loop runs), but not with the code on `dataseed` that updates the `mv` matrix first, and only then applies all of its updates to the `y` matrix.

IMHO, neither of these impls is ideal; it seems like what we really want here is a multinomial distribution over the dests for a given ori, right? Either way, I'm happy to update this to keep the `mv` struct intact, even though I won't actually apply it to the `y` matrix until a little later in the function call to save a loop.

Reply:
We indeed need to dicuss mobility integration (see issue #90 )

Reply:
I think we do ideally want the multinomial as a short term solution.  However, for it to be effective, we would need to use row sparse matrices instead of coo matrices.  I'm going to accept this PR for now, since the current code is definitely broken.  However, we shouldn't think of this as resolving this issue.  Really we need to do something to deal with mobility in a more principled way.